sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -857,6 +875,37 @@ static struct snd_soc_dai_link *sof_card_dai_links_create(struct device *dev,
 		links[id].num_cpus = 1;
 	}
 
+	/* HDMI-In SSP */
+	if (sof_rt5682_quirk & SOF_SSP_HDMI_CAPTURE_PRESENT) {
+		int num_of_hdmi_ssp = (sof_rt5682_quirk & SOF_NO_OF_HDMI_CAPTURE_SSP_MASK) >>
+				SOF_NO_OF_HDMI_CAPTURE_SSP_SHIFT;
+
+		for (i = 1; i <= num_of_hdmi_ssp; i++) {
+			int port = (i == 1 ? (sof_rt5682_quirk & SOF_HDMI_CAPTURE_1_SSP_MASK) >>
```

Comment:
Here its not hard coded SSP port, to identify and mask the respective HDMI capture SSP port by this index. it helps to select the mask flag which need to be used among the two HDMI capture port case.

Reply:
If num_of_hdmi_ssp == 1, then you will only use SSP1_MASK. This sort of code is not great really, it's a mix of variable parts and fixed ones.

Reply:
Actually the loop is used to create the links for number of HDMI capture I2S ports and fixed condition used to identify and mask the respective SSP port number. i.e. to select either HDMI1 or 2. 
otherwise it will lead into nested if else condition. Please suggest if you have any other optimized approach.

Reply:
Ah yes, it's capture_1, not ssp_1. Fine.